src/components/signUpInOut/SignUpForm.jsx

```diff
@@ -0,0 +1,149 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import {
+  Form, Button, Row, Col, Modal,
+} from 'react-bootstrap';
+import { FaXmark } from 'react-icons/fa6';
+import { signUp } from '../../features/authSlice';
+
+const initialAddressData = {
+  street: '',
+  city: '',
+  state: '',
+  zip_code: '',
+};
+
+const initialFormData = {
+  name: '',
+  age: '',
+  email: '',
+  photo: '',
+  role: '',
+  password: '',
+  password_confirmation: '',
+  address: { ...initialAddressData },
+};
+
+const SignUpForm = ({ showSignUpModal, handleCloseSignUpModal, handleOpenSignUpModal }) => {
+  const [formData, setFormData] = useState({ ...initialFormData });
+  const dispatch = useDispatch();
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevFormData) => {
+      if (name.startsWith('address.')) {
+        const addressField = name.split('.')[1];
+        return {
+          ...prevFormData,
+          address: {
+            ...prevFormData.address,
+            [addressField]: value,
+          },
+        };
+      }
+      return {
+        ...prevFormData,
+        [name]: value,
+      };
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(signUp(formData));
+    window.location.href = '/';
+  };
+
+  return (
```

Comment:
- [ ] I cannot delete doctors:

![image](https://github.com/baqar-abbas/appointment-app-front-end/assets/57421823/4f7c7b26-30cf-45e6-bf3d-d7d1615bebc4)

![image](https://github.com/baqar-abbas/appointment-app-front-end/assets/57421823/91314ec4-0756-4f94-ab53-b4864742493d)

![image](https://github.com/baqar-abbas/appointment-app-front-end/assets/57421823/5f3bcb14-ec8c-4839-90c3-bee1d3052fb1)

![image](https://github.com/baqar-abbas/appointment-app-front-end/assets/57421823/3374f7ae-c5df-4d83-ae4e-6e0025a320cf)

Suggestion:  Doctor deletion is not a requirement for groups of 2 people.  If you can't find the bug, you can remove this deletion functionality.

Reply:
Thanks for the example users on the live demo.  All seems to work well with those users, except for the fact that the admin can't delete doctors.

![image](https://github.com/baqar-abbas/appointment-app-front-end/assets/57421823/f5fb7312-f0f2-4fc0-b7aa-cf4f6d047d6d)

Suggestion/reminder: The deletion functionality is not a requirement for groups of 2 people.  But if it's there, it needs to work.

Reply:
Delete doctor issue has been fixed. Screen shots attached below for your kind reference.

I can add doctor by logging in with Admin account (Please check these changes locally first using rails backend api http:localhost:3000/.... Once you approved changes can be incorporated on deploymemt on render.
![addCreateDoctor](https://github.com/baqar-abbas/appointment-app-front-end/assets/117091923/d05a7532-6b1a-45af-8983-a5b0a6ca5362)
![doctorAdded](https://github.com/baqar-abbas/appointment-app-front-end/assets/117091923/9b288ee0-2282-4e9a-be96-8a39fcd580be)

I can also delete doctor by logging in with Admin account (these changes will be deployed on deployment at render once you approve please check these changes locally using local host link) 
Screen shot attached for your kind reference.

![deleteDoctor](https://github.com/baqar-abbas/appointment-app-front-end/assets/117091923/a95ab1fd-33ab-40de-b70d-d03e626e5c2f)
![doctorDeleted](https://github.com/baqar-abbas/appointment-app-front-end/assets/117091923/3b3022ae-268b-4d96-8f63-0d9ba8ad22a9)

All suggested changes have been implemented sending it for review again.

Reply:
Please Note: you can test these implemented changed locally using rails backend once approved I can update the deployment on render as well. Thanks